src/bun.js/test/jest.zig

```diff
@@ -364,24 +363,277 @@ pub const Expect = struct {
         active_test_expectation_counter.actual += 1;
 
         const expected = arguments[0].coerce(i32, globalObject);
-        const value = JSC.Jest.Expect.capturedValueGetCached(thisValue) orelse {
+        const value: JSValue = JSC.Jest.Expect.capturedValueGetCached(thisValue) orelse {
             globalObject.throw("Internal consistency error: the expect(value) was garbage collected but it should not have been!", .{});
             return .zero;
         };
-        const actual = value.getLengthOfArray(globalObject);
-        if (expected != actual) {
-            globalObject.throw("Expected length to equal {d} but received {d}\n  Expected: {d}\n    Actual: {d}\n", .{
-                expected,
-                actual,
-                expected,
-                actual,
-            });
+        value.ensureStillAlive();
+
+        const not = this.op.contains(.not);
+        var pass = false;
+        const actual_length = value.getLengthOfArray(globalObject);
+        if (actual_length == expected) pass = true;
+
+        if (not) pass = !pass;
+        if (pass) return thisValue;
+
+        // handle failure
+        if (not) {
+            globalObject.throw("\n\tExpected: not length {d}\n\tReceived: {d}", .{ expected, actual_length });
+        } else {
+            globalObject.throw("\n\tExpected: length {d}\n\tReceived: {d}", .{ expected, actual_length });
+        }
+        return .zero;
+    }
+
+    pub fn toContain(
+        this: *Expect,
+        globalObject: *JSC.JSGlobalObject,
+        callFrame: *JSC.CallFrame,
+    ) callconv(.C) JSC.JSValue {
+        const thisValue = callFrame.this();
+        const arguments_ = callFrame.arguments(1);
+        const arguments = arguments_.ptr[0..arguments_.len];
+
+        if (arguments.len < 1) {
+            globalObject.throwInvalidArguments("toContain() takes 1 argument", .{});
             return .zero;
         }
 
-        return thisValue;
+        if (this.scope.tests.items.len <= this.test_id) {
+            globalObject.throw("toContain() must be called in a test", .{});
+            return .zero;
+        }
+
+        active_test_expectation_counter.actual += 1;
+
+        const expected = arguments[0];
+        expected.ensureStillAlive();
+        const value: JSValue = JSC.Jest.Expect.capturedValueGetCached(thisValue) orelse {
+            globalObject.throw("Internal consistency error: the expect(value) was garbage collected but it should not have been!", .{});
+            return .zero;
+        };
+        value.ensureStillAlive();
+
+        const not = this.op.contains(.not);
+        var pass = false;
+
+        if (value.isIterable(globalObject)) {
```

Comment:
Should this check for strings first? Strings are iterable too, at least in JS

Reply:
this is valid in Jest:
```js
const s1 = new String("123");
expect(s1).not.toContain("12");
const s2 = "123";
expect(s2).toContain("12");
```
in this situation `isIterable` returns false for `"123"` and true for `new String("123")`